test/mocks/init/mocks.h

```diff
@@ -50,6 +50,13 @@ class ExpectableTargetImpl : public TargetImpl {
   ::testing::internal::TypedExpectation<void()>& expectInitializeWillCallReady();
 };
 
+class ExpectableSharedTargetImpl : public SharedTargetImpl {
+public:
+  ExpectableSharedTargetImpl(absl::string_view name = "test");
+  ExpectableSharedTargetImpl(absl::string_view name, InitializeFn fn);
+  int count_{0};
```

Comment:
This probably isn't how you want to implement this. Keep in mind that if you have a mock method, you can specify how many times you expect it to be called. See `ExpectableTargetImpl` above.

Reply:
Good point. I always forgot the weapon of define and mock new methods. Let me give it a try

Reply:
I borrow the ExpectableTargetImpl by pure renaming. However, I find it not very fitting. See my comment in the outdated commit for the demonstration

Reply:
+1 on what @mergeconflict suggests. The rest of the PR LGTM.